pkg/daemon/ceph/osd/daemon.go

```diff
@@ -415,6 +415,25 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 				}
 				matchedDevice = desiredDevice
 
+				if agent.pvcBacked {
+					crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
+					if crushDeviceClass != "" {
+						matchedDevice.DeviceClass = crushDeviceClass
+					}
+				}
+
+				if matchedDevice.DeviceClass == "" {
+					if device.Rotational {
```

Comment:
Look like the code below https://github.com/rook/rook/pull/8028/files#diff-c6bc098c9a4e7a6d3ee38e354bae71cc5e9a5421211f1d472f0a17542b178851R1019-R1027 has a similar logic, can we factor this into a helper function? Also one uses `strings.Contains` and the other `strings.HasPrefix` I think we agreed on `strings.Contains`.

Reply:
The thing is in https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/osd.go#L452 we are using RealPath like `/dev/nvme1e` and in current place we are using device.Name which is just `nvme` - but i think i can wrap it somehow into helper function